compiler/ast.nim

```diff
@@ -1591,7 +1591,7 @@ proc propagateToOwner*(owner, elem: PType; propagateHasAsgn = true) =
   if mask != {} and propagateHasAsgn:
     let o2 = owner.skipTypes({tyGenericInst, tyAlias, tySink})
     if o2.kind in {tyTuple, tyObject, tyArray,
-                   tySequence, tySet, tyDistinct, tyOpenArray, tyVarargs}:
+                   tySequence, tySet, tyDistinct, tyVarargs}:
```

Comment:
```nim
  let mask = elem.flags * {tfHasAsgn, tfHasOwned}
  if mask != {} and propagateHasAsgn:
    let o2 = owner.skipTypes({tyGenericInst, tyAlias, tySink})
    if o2.kind in {tyTuple, tyObject, tyArray,
                   tySequence, tySet, tyDistinct, tyVarargs}:
      o2.flags.incl mask
      owner.flags.incl mask
```
Before if `openArray` contains seq/string etc, its flags will include `tfHasAsgn` too.

And here is the definition of `hasDestructor`.
```
template hasDestructor*(t: PType): bool = {tfHasAsgn, tfHasOwned} * t.flags != {}
```

Reply:
As a bonus, I find another bug
```
proc setPosition(params: varargs[string]) =
    for i in params.toOpenArray(0, params.len - 1):
        echo i

proc uciLoop*() =
    let params = @[readLine(stdin)]
    setPosition(params)

uciLoop()
```
gives
```
Error: internal error: '=destroy' operator not found for type openArray[string]
```

Reply:
Also remove the `tyVarargs` case, varargs's runtime behavior is identical to openArray.

Reply:
yeah, I guess it is less harmful for `tyVarargs` becuase it cannot be used as return types.